### PR TITLE
Make FormParams have the correct DWARF version.

### PR DESCRIPTION
**Pull request by rastogishubham**
This PR addresses this comment https://github.com/apple/llvm-project/pull/7852#discussion_r1427906235

### CODE REVIEW

**Comment by rastogishubham** [comment]
@swift-ci please test llvm

**Comment by rastogishubham** [comment]
@swift-ci please test

File: llvm/lib/MCCAS/MCCASObjectV1.cpp [hunk 1]
```diff
@@ -3143,8 +3144,8 @@ Error DIEVisitor::visitDIEAttrs(BinaryStreamReader &DataReader,
                                 ArrayRef<AbbrevContent> DIEContents) {
   constexpr auto IsLittleEndian = true;
   constexpr auto AddrSize = 8;
-  constexpr auto FormParams =
-      dwarf::FormParams{4 /*Version*/, AddrSize, dwarf::DwarfFormat::DWARF32};
+  auto FormParams = dwarf::FormParams{DwarfVersion /*Version*/, AddrSize,
```
Comment (felipepiovezan) [inline llvm/lib/MCCAS/MCCASObjectV1.cpp]: I guess we don't need the inline comment anymore :)

File: llvm/lib/MCCAS/MCCASObjectV1.cpp [hunk 2]
```diff
@@ -3254,8 +3255,8 @@ static std::optional<uint8_t> getNonULEBFormSize(dwarf::Form Form,
 
 Error DIEVisitor::materializeAbbrevDIE(unsigned AbbrevIdx) {
   constexpr auto AddrSize = 8;
-  constexpr auto FormParams =
-      dwarf::FormParams{4 /*Version*/, AddrSize, dwarf::DwarfFormat::DWARF32};
+  auto FormParams = dwarf::FormParams{DwarfVersion /*Version*/, AddrSize,
```
Comment (felipepiovezan) [inline llvm/lib/MCCAS/MCCASObjectV1.cpp]: same here!

**Review by felipepiovezan — APPROVED** [review]

**Comment by rastogishubham** [comment]
@swift-ci please test llvm

**Pull request merged**